src/main/java/org/mapstruct/intellij/util/MapstructUtil.java

```diff
@@ -200,17 +201,35 @@ private static boolean isPublic(@NotNull PsiField field) {
 
     public static boolean isPublicModifiable(@NotNull PsiField field) {
         return isPublicNonStatic( field ) &&
-               !field.hasModifierProperty( PsiModifier.FINAL );
+            !field.hasModifierProperty( PsiModifier.FINAL );
     }
 
     public static boolean isFluentSetter(@NotNull PsiMethod method, PsiType psiType) {
         return !psiType.getCanonicalText().startsWith( "java.lang" ) &&
             method.getReturnType() != null &&
             !isAdderWithUpperCase4thCharacter( method ) &&
-            TypeConversionUtil.isAssignable(
-                psiType,
-                PsiUtil.resolveGenericsClassInType( psiType ).getSubstitutor().substitute( method.getReturnType() )
-            );
+            isAssignableFromReturnTypeOrSuperTypes( psiType, method.getReturnType() );
+    }
+
+    private static boolean isAssignableFromReturnTypeOrSuperTypes(PsiType psiType, @Nullable PsiType returnType) {
+
+        if ( returnType == null ) {
+            return false;
+        }
+
+        if ( isAssignableFrom( psiType, returnType ) ) {
+            return true;
+        }
+
+        return Arrays.stream( returnType.getSuperTypes() )
+            .anyMatch( superType -> isAssignableFrom( psiType, superType ) );
```

Comment:
There is no need to create a stream and then use `anyMatch`. Let's just iterate through the `returnType.getSuperTypes()` and do a check in the for loop.